lib/tests/streamlit/elements/doc_string_test.py

```diff
@@ -33,6 +33,20 @@ def patch_varname_getter():
     )
 
 
+class ConditionalHello:
+    def __init__(self, available):
+        self.available = available
+
+    def __getattribute__(self, name):
+        if name == "say_hello" and not self.available:
+            raise AttributeError(f"{name} is not accessible when x is even")
+        else:
+            return object.__getattribute__(self, name)
+
+    def say_hello(self):
+        print("Hello")
```

Comment:
super nit: doesn't really matter since this method is never called, but it might be slightly preferable to have the body of this method be a no-op to not have a print statement in the code